lib/Parser/Parse.cpp

```diff
@@ -1682,10 +1682,12 @@ void Parser::CreateSpecialSymbolDeclarations(ParseNodePtr pnodeFnc, bool isGloba
 {
     Assert(!(isGlobal && (this->m_grfscr & fscrEval)));
 
+    bool isTopLevelEventHandler = (this->m_grfscr & fscrImplicitThis || this->m_grfscr & fscrImplicitParents);
```

Comment:
This is actually too conservative. We only have the implicit "this" reference in the top lexical level of an event handler. Functions declared inside the enclosing function don't need this.

Reply:
Thanks, Paul. We made this change yesterday to try and unblock the DRT. 

Is the event handler treated like global code?